pkg/roachpb/api.proto

```diff
@@ -1432,6 +1432,7 @@ message AddSSTableRequest {
 
   RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
   bytes data = 2;
+  bool disallow_shadowing = 3;
```

Comment:
this could use a comment explaining what it does. I might also go ahead and fold this into the 3rd commit (that implements it in batcheval and tests it).